### DISCUSSION TITLE
React stateless components?

### DISCUSSION THREAD

**Discussion by emilebosch** [ruby2js/ruby2js#130] [Ideas]
Hi all! I love react/preact's stateless components what would be needed to be done to enable this?

```ruby
Button = ->(x) { %x(<button>{x}</button>)}
```
Would be a great addition!

**Comment by rubys** [comment]
Given just that code, I don't see a way to determine whether to generate a React component or a Preact component.  :-(

**Reply by rubys** [reply]
I've pushed a change, assuming for now that React is what is intended: https://github.com/ruby2js/ruby2js/commit/87603af79b6683975b948218c453b2ce533b69ce

**Reply by emilebosch** [reply]
Ah I meant more in general. That lambdas can be used to generate stateless components in React. Right now the sample  code doesn't generate the equivalent of a react stateless component, and only components that have `render` generate proper react components. But maybe I'm missing something? 

Whether to generate react or react would be determined by a filter I think?

**Reply by emilebosch** [reply]
Hahaha, you wrote a change faster than I replied. What is this? Magic? Haha! good work!
